server/api_server/schemas.py

```
@@ -0,0 +1,625 @@
+    return value.upper()
+
+
+def validate_ip(value: str) -> str:
+    """Validate IP address format (IPv4 or IPv6) using stdlib ipaddress.
+
+    Returns the canonical string form of the IP address.
+    """
+    try:
+        ip = ipaddress.ip_address(value)
+    except ValueError:
+        raise ValueError(f"Invalid IP address format: {value}")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)